internal/datastreams/transport.go

```diff
@@ -99,12 +99,12 @@ func (t *httpTransport) sendPipelineStats(p *StatsPayload) error {
 	if err != nil {
 		return err
 	}
+	msg := make([]byte, 1000)
+	n, _ := resp.Body.Read(msg)
+	resp.Body.Close()
 	if code := resp.StatusCode; code >= 400 {
 		// error, check the body for context information and
 		// return a nice error.
```

Comment:
```suggestion
	defer resp.Body.Close()
	if code := resp.StatusCode; code >= 400 {
		// error, check the body for context information and
		// return a nice error.
		msg := make([]byte, 1000)
		n, _ := resp.Body.Read(msg)
```

If there's a chance you're going to need to throw away `n`, then don't read it until you know you need to. I suggest just deferring the close and then reading the body if and when you really need it

Reply:
So, ideally I would like to do that, but it looks like we need to read the bytes for the structure to be re-used:
```
// Body which the user is expected to close. If the Body is not both
// read to EOF and closed, the Client's underlying RoundTripper
// (typically Transport) may not be able to re-use a persistent TCP
// connection to the server for a subsequent "keep-alive" request.
```

From here:
https://github.com/golang/go/blob/master/src/net/http/client.go#L562

Reply:
This doesn't guarantee we read to EOF, just that we read <=1000 bytes.

Let's just use the standard `io.Copy` and `io.Discard` like we do in the agent: https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/api/api.go#L467
ideally with a defer along with the Close.

Then we can only allocate a buffer for the message and read into it if we actually need it.